mock.go

```diff
@@ -271,6 +272,9 @@ func (k key) Less(other key) bool {
 		if cmp == 0 {
 			continue
 		}
+		if k[i].ClusteringOrder { // desc
```

Comment:
It would be great if we could add a test to the mock exercising this behaviour. This can be added as a subsequent PR though if you want to get this merged 🙏